### PR TITLE
Disk Based Collection Partition 

### PR DESCRIPTION
**Pull request by chathurawidanage**

### CODE REVIEW

File: twister2/data/src/main/java/edu/iu/dsc/tws/dataset/partition/DiskBackedCollectionPartition.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  private MessageType dataType = MessageTypes.OBJECT;
+
+  private List<Path> filesList = new ArrayList<>();
+  private Path tempDirectory;
+
+  public DiskBackedCollectionPartition(int maxFramesInMemory) {
+    super();
+    this.maxFramesInMemory = maxFramesInMemory;
+    try {
+      this.tempDirectory = Files.createTempDirectory(UUID.randomUUID().toString());
```
Comment (supunkamburugamuve) [inline twister2/data/src/main/java/edu/iu/dsc/tws/dataset/partition/DiskBackedCollectionPartition.java]: We should write to the volatile volume

Comment (nirandaperera) [reply twister2/data/src/main/java/edu/iu/dsc/tws/dataset/partition/DiskBackedCollectionPartition.java]: volatile volume is null in the standalone distribution IINM

Comment (supunkamburugamuve) [reply twister2/data/src/main/java/edu/iu/dsc/tws/dataset/partition/DiskBackedCollectionPartition.java]: I'm not sure how these will work with K8s. Maybe @ahmet-uyar can comment?

Comment (ahmet-uyar) [reply twister2/data/src/main/java/edu/iu/dsc/tws/dataset/partition/DiskBackedCollectionPartition.java]: I don't know the details of this implementation but in K8s, we have a local volatile volume that is based on the local disk of the machine that a worker is running on. We also have shared NFS directory. That is accessible by all workers in a job.

Comment (supunkamburugamuve) [reply twister2/data/src/main/java/edu/iu/dsc/tws/dataset/partition/DiskBackedCollectionPartition.java]: I think this issue can be related to shuffling as well. Created an issue to track this. Let's merge this and fix that issue separately

https://github.com/DSC-SPIDAL/twister2/issues/662

**Pull request merged**